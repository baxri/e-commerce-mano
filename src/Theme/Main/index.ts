import button from "./button";
import colors from "./colors";
import common from "./common";
import fonts from "./fonts";

export default {
  name: "Main Theme",
  dark: false,
  button,
  common,
  colors,
  fonts,

  // --   0  1  2  3  4   5   6   7   8   9
  space: [0, 2, 4, 8, 12, 16, 20, 24, 32, 64],
  // --       0   1   2   3    4   5   7   8   9
  fontSizes: [10, 12, 14, 16, 18, 20, 22, 24, 26],
  // --         0   1   2   3    4   5   7   8   9
  lineHeights: [13, 15, 17, 19, 21, 23, 25, 27, 29],
  // --         0    1    2    3
  fontWeights: [400, 500, 600, 700],
};
