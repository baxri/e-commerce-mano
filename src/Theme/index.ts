import MainTheme from "./Main";

enum ThemeID {
  Main = "main",
}

const THEMES: Record<ThemeID, any> = {
  [ThemeID.Main]: MainTheme,
};

export { ThemeID, THEMES };
