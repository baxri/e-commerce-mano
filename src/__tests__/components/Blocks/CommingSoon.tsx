import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { ThemeID, THEMES } from 'Theme';

import CommingSoon from 'Components/Blocks/CommingSoon';

test('SommingSoon', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={THEMES[ThemeID.Main]}>
        <CommingSoon />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
