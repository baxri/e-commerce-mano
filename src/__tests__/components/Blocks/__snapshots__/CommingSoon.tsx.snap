// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SommingSoon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "display": "flex",
      "flexDirection": "column",
      "maxWidth": 200,
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    color="grey"
    fontFamily="Inter"
    fontSize={2}
    fontWeight={3}
    mt={8}
    style={
      [
        {
          "color": "#6D758E",
          "fontFamily": "Inter",
          "fontSize": 14,
          "fontWeight": "700",
          "marginTop": 32,
        },
      ]
    }
  >
    Coming soon
  </Text>
  <Text
    allowFontScaling={false}
    color="grey"
    fontFamily="Inter"
    fontSize={1}
    fontWeight={0}
    ml={3}
    mt={5}
    style={
      [
        {
          "color": "#6D758E",
          "fontFamily": "Inter",
          "fontSize": 12,
          "fontWeight": "400",
          "marginLeft": 8,
          "marginTop": 16,
        },
      ]
    }
  >
    Come back to this page, this feature will be available soon
  </Text>
</View>
`;
